Tasks/AppStorePromote/Tests/L0.ts

```diff
@@ -116,6 +117,74 @@ describe('app-store-promote L0 Suite', function () {
         done();
     });
 
+    it('service endpoint with api key', (done: Mocha.Done) => {
+        this.timeout(1000);
+
+        let tp = path.join(__dirname, 'L0ApiKeyEndPoint.js');
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        const buildPath = 'test_build_path';
+        const keyFilePath = path.join(buildPath, 'api_keyD383SF739.json');
+
+        if (!fs.existsSync(buildPath)) {
+            fs.mkdirSync(buildPath);
+        }
+
+        tr.run();
+        assert(tr.ran(`fastlane deliver submit_build --precheck_include_in_app_purchases false --api_key_path ${keyFilePath} -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true --force`), 'fastlane deliver with api key should have been run.');
+        assert(tr.invokedToolCount === 1, 'should have run only fastlane deliver.');
+        assert(tr.succeeded, 'task should have succeeded');
+
+        assert(fs.existsSync(keyFilePath), 'api_key.json file should have been created');
+
+        let rawdata = fs.readFileSync(keyFilePath, 'utf8');
+        let apiKey = JSON.parse(rawdata);
+
+        assert(apiKey.key_id === 'D383SF739', 'key_id should be correct');
+        assert(apiKey.issuer_id === '6053b7fe-68a8-4acb-89be-165aa6465141', 'issuer_id should be correct');
+        assert(apiKey.key === 'LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1JR1RBZ0VBTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEJIa25saGRsWWRMdQotLS0tLUVORCBQUklWQVRFIEtFWS0tLS0t', 'key should be correct');
+        assert(apiKey.in_house === false, 'in_house should be correct');
+        assert(apiKey.is_key_content_base64 === true, 'is_key_content_base64 should be correct');
+
+        fs.unlinkSync(keyFilePath);
+        fs.rmdirSync(buildPath);
```

Comment:
These cleanup lines don't get executed if any assertions above fail. This leaves the `api_keyD383SF739.json` file in the repository. @jfahrenkrug could you please take a look at this issue?

Reply:
@DaniilShmelev Great catch, thanks, looking at it now.

Reply:
@DaniilShmelev fixed in https://github.com/microsoft/app-store-vsts-extension/pull/189/commits/45ce07305b71d8ed95a3634a95253ed257c5db56